packages/root-cms/ui/components/RichTextEditor/lexical/LexicalEditor.tsx

```diff
@@ -0,0 +1,184 @@
+import './LexicalEditor.css';
+
+import {AutoLinkNode, LinkNode} from '@lexical/link';
+import {ListItemNode, ListNode} from '@lexical/list';
+import {
+  InitialConfigType,
+  LexicalComposer,
+} from '@lexical/react/LexicalComposer';
+import {useLexicalComposerContext} from '@lexical/react/LexicalComposerContext';
+import {ContentEditable} from '@lexical/react/LexicalContentEditable';
+import {LexicalErrorBoundary} from '@lexical/react/LexicalErrorBoundary';
+import {HistoryPlugin} from '@lexical/react/LexicalHistoryPlugin';
+import {LinkPlugin} from '@lexical/react/LexicalLinkPlugin';
+import {ListPlugin} from '@lexical/react/LexicalListPlugin';
+import {RichTextPlugin} from '@lexical/react/LexicalRichTextPlugin';
+import {TabIndentationPlugin} from '@lexical/react/LexicalTabIndentationPlugin';
+import {HeadingNode} from '@lexical/rich-text';
+import {useMemo, useState} from 'preact/hooks';
+import {RichTextData} from '../../../../shared/richtext.js';
+import {joinClassNames} from '../../../utils/classes.js';
+import {
+  SharedHistoryProvider,
+  useSharedHistory,
+} from './hooks/useSharedHistory.js';
+import {ToolbarProvider} from './hooks/useToolbar.js';
+import {LexicalTheme} from './LexicalTheme.js';
+import {FloatingLinkEditorPlugin} from './plugins/FloatingLinkEditorPlugin.js';
+import {FloatingToolbarPlugin} from './plugins/FloatingToolbarPlugin.js';
+import {MarkdownTransformPlugin} from './plugins/MarkdownTransformPlugin.js';
+import {OnChangePlugin} from './plugins/OnChangePlugin.js';
+import {ShortcutsPlugin} from './plugins/ShortcutsPlugin.js';
+import {ToolbarPlugin} from './plugins/ToolbarPlugin.js';
+
+const INITIAL_CONFIG: InitialConfigType = {
+  namespace: 'RootCMS',
+  theme: LexicalTheme,
+  nodes: [
+    AutoLinkNode,
+    HeadingNode,
+    // ImageNode,
```

Comment:
I've seen a few projects that use the EditorJs image embed; are they incompatible with Lexical at this time?

Reply:
it's compatible but it requires a custom upload dialog from us which i haven't written yet.